Specify mysql should be installed as part of development bootstrap

es opened the pull request:
After following the setup guide in `DEVELOPMENT.md`, I still ran into an issue running tests:
```
ERRO[0000] Error: exec: "mysql": executable file not found in $PATH
```

Turns out `mysql` must be installed locally for `buffalo` to work. This PR adds that dependency to the bootstrap guide.

robjloranger commented:
Can you open an issue to discuss your problem? I just removed mysql and ran `make dev` with success. We use docker for database services and so should not require mysql to be installed.

robjloranger commented:
double checked and I can run both Olympus and Zeus with no installation of mysql via `buffalo dev`

arschles commented:
I can’t confirm that bc I’m pretty sure it is because buffalo test uses it
under the hood. We could do some work though to remove that as a
requirement if we wanted. That test command used the mysql cli to run
migrations but we could do that in code. What do you think?

On Tue, Jul 24, 2018 at 17:24 Rob j Loranger <notifications@github.com>
wrote:

> @arschles <https://github.com/arschles> can you confirm that mysql should
> not be an installation requirement and close this if so?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gomods/athens/pull/323#issuecomment-407571710>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAEU0RcB-5mfgXi-rswU3_RF9qe_g4tmks5uJ56EgaJpZM4Vbzmy>
> .
>

robjloranger commented:
@arschles in that case either or is fine by me. Merging this or getting rid of the requirement. How can I invoke the test to verify?

arschles commented:
Sorry I didn’t catch - which part are you trying to verify?

On Tue, Jul 24, 2018 at 19:21 Rob j Loranger <notifications@github.com>
wrote:

> @arschles <https://github.com/arschles> in that case either or is fine by
> me. Merging this or getting rid of the requirement. How can I invoke the
> test to verify?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gomods/athens/pull/323#issuecomment-407611839>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAEU0VnA-nM7GlDKzHdtA_zhRfnX6meoks5uJ9YMgaJpZM4Vbzmy>
> .
>

robjloranger commented:
Just that mysql is in fact required. I have run our make dev script and also run both development app servers without it so just wondering what might be looking for it. What script/test etc is trying to use mysql. Is it just the tests for our packages? @marpio mentioned it was an old requirement so wanted to be certain before we add it as a hard installation requirement.

marpio commented:
The gobuffalo/pop PR which removes the mysql client dependency for schema loading: https://github.com/gobuffalo/pop/pull/125

es commented:
It looks like @marpio was right. I was running `v0.11.0` - once I updated to `v0.12.4`, the tests ran fine without MySQL client installed locally. 

I've modified the PR to specify buffalo v0.12.4 or higher is required.

arschles commented:
Thank you! 😊

On Fri, Jul 27, 2018 at 13:17 Rob j Loranger <notifications@github.com>
wrote:

> Merged #323 <https://github.com/gomods/athens/pull/323>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gomods/athens/pull/323#event-1758063005>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AAEU0fYxIawoxOZ3oQMjounqyiXeTMSyks5uK3VMgaJpZM4Vbzmy>
> .
>
